accounts-on-demand/src/subscription_manager.rs

```diff
@@ -0,0 +1,281 @@
+    SubscribeRequestFilterAccountsFilterMemcmp,
+};
+
+pub struct SubscriptionManger {
+    accounts_filters: Arc<RwLock<AccountFilters>>,
+    restart_sender: broadcast::Sender<()>,
+}
+
+impl SubscriptionManger {
+    pub fn new(
+        grpc_sources: Vec<GrpcSourceConfig>,
+        accounts_storage: Arc<dyn AccountStorageInterface>,
+        account_notification_sender: Sender<AccountNotificationMessage>,
+    ) -> Self {
+        let (restart_sender, recver) = broadcast::channel(1);
+        let accounts_filters = Arc::new(RwLock::new(vec![]));
+
+        let (_, mut account_stream) =
+            create_grpc_account_streaming(grpc_sources, accounts_filters.clone(), recver);
+
+        let task_restart_sender = restart_sender.clone();
+        tokio::spawn(async move {
+            loop {
+                match tokio::time::timeout(Duration::from_secs(60), account_stream.recv()).await {
+                    Ok(Ok(message)) => {
+                        let _ = account_notification_sender.send(message.clone());
+                        accounts_storage
+                            .update_account(message.data, message.commitment)
+                            .await;
+                    }
+                    Ok(Err(e)) => match e {
+                        broadcast::error::RecvError::Closed => {
+                            log::error!("Account stream channel closed");
+                            break;
+                        }
+                        broadcast::error::RecvError::Lagged(_) => {
+                            continue;
+                        }
+                    },
+                    Err(_) => {
+                        let _ = task_restart_sender.send(());
+                    }
+                }
+            }
+        });
+        Self {
+            accounts_filters,
+            restart_sender,
+        }
+    }
+
+    pub async fn update_subscriptions(&self, filters: AccountFilters) {
+        *self.accounts_filters.write().await = filters;
+        let _ = self.restart_sender.send(());
+    }
+}
+
+pub fn start_account_streaming_tasks(
+    grpc_config: GrpcSourceConfig,
+    accounts_filters: Arc<RwLock<AccountFilters>>,
+    mut restart_channel: broadcast::Receiver<()>,
+    account_stream_sx: broadcast::Sender<AccountNotificationMessage>,
+) -> AnyhowJoinHandle {
+    tokio::spawn(async move {
+        'main_loop: loop {
+            let processed_commitment = yellowstone_grpc_proto::geyser::CommitmentLevel::Processed;
+
+            let mut subscribe_accounts: HashMap<String, SubscribeRequestFilterAccounts> =
+                HashMap::new();
+
+            let accounts_filters = accounts_filters.read().await.clone();
+            for (index, accounts_filter) in accounts_filters.iter().enumerate() {
+                if !accounts_filter.accounts.is_empty() {
+                    subscribe_accounts.insert(
+                        format!("accounts_{index:?}"),
```

Comment:
I like that you prefix the filter keys but could you use a dedicated prefix for accounts and programs?
`account_0` and `program_account_12121221`